src/main/kotlin/com/ampnet/reportserviceeth/blockchain/BlockchainEventServiceImpl.kt

```diff
@@ -30,6 +30,10 @@ class BlockchainEventServiceImpl(
     private val chainPropertiesHandler: ChainPropertiesHandler
 ) : BlockchainEventService {
 
+    private data class CacheKey(val address: String, val chainId: Long)
+    private val assetCache: MutableMap<CacheKey, IAssetCommon.AssetCommonState> = mutableMapOf()
```

Comment:
It would be better to use `ConcurrentHashMap` since this map could be used by multiple threads at once (one thread per chain if I'm not mistaken). Regular maps can potentially have some concurrency issues when hash collision in the map occurs. I think that by default maps use 64 buckets to store data, so hash collisions will occur quite often.